### PR TITLE
chore: remove sha [DET-5225]

### PR DESCRIPTION
**Pull request by stoksc**
## Description
This change removes the SHA, adaptive, and adaptive_simple searchers. They have been officially deprecated since [0.13.7](https://docs.determined.ai/latest/release-notes.html#version-0-13-7).

## Test Plan
- [x] make sure tests still pass

## Commentary (optional)
Still need to fix new expconf packages

## Checklist

- [x] User-facing API changes need the "User-facing API Change" label.
- [x] Release notes should be added as a separate file under `docs/release-notes/`.
See [Release Note](../docs/release-notes/README.md) for details.
- [x] Licenses should be included for new code which was copied and/or modified from any external code.

[DET-1234]: https://determinedai.atlassian.net/browse/DET-1234

### CODE REVIEW

**Comment by stoksc** [comment]
blocked on #2180

File: docs/topic-guides/hp-tuning-det/hp-adaptive-asha.txt
Comment (stoksc) [inline docs/topic-guides/hp-tuning-det/hp-adaptive-asha.txt:52]: @liamcli this is a mash-up of all the previous adaptive guides, but rewritten a bit to remove the stuff that was wrong from sha to asha. i removed the old adaptive guides. if you could, either review this section or just take a pass after this lands to add corrections (that just may be faster than you reviewing it old, or honestly push some fixes to my branch if you want)

**Review by rb-determined-ai — COMMENTED** [review]
I found some more references to old searchers:

* `docs/topic-guides/training/reproducibility.txt` talks about `adaptive`
* `docs/topic-guides/training/training-units.txt` talks about "adaptive searchers"
* `docs/tutorials/quick-start.txt` references `adaptive_simple`
* `webui/react/src/fixtures/responses/experiment-details/set-a.json` has several `adaptive` and `adaptive_simpe` experiments
* `webui/react/src/types.ts` the webui has its own experiment config parsing

(I used recursive gnu-grep with perl syntax for negative lookaheads: `grep -rnP 'adaptive(?!.asha)' .`)

File: docs/topic-guides/hp-tuning-det/hp-adaptive-asha.txt
Comment (stoksc) [reply docs/topic-guides/hp-tuning-det/hp-adaptive-asha.txt:167]: Yeah true, though i fixed it up but it's still not quite right. I'll fix it better.

**Comment by stoksc** [comment]
* `docs/topic-guides/training/reproducibility.txt` ✅ 
* `docs/topic-guides/training/training-units.txt` These link to the correct docs but I'll change adaptive to adaptive asha since it's the only now.
* `docs/tutorials/quick-start.txt` this definitely is a mistake, since the quick start ref's an `adaptive_asha` searcher ✅ 
* `webui/react/src/fixtures/responses/experiment-details/set-a.json` seems to be a fixture for a test, I think it's important the webui still loads old searchers so i'll leave this in the fixture
* `webui/react/src/types.ts` ditto, i'll leave webui stuff alone, but cc @hkang1 that sha is going away.

**Comment by rb-determined-ai** [comment]
Wait, if we say that the webui should be able to view old experiments... will we hit errors unmarshaling old configs from the database?  Will the master be able to serve up any info at all for these old experiments?

**Comment by stoksc** [comment]
> Wait, if we say that the webui should be able to view old experiments... will we hit errors unmarshaling old configs from the database? Will the master be able to serve up any info at all for these old experiments?

should be fine, hit a lot of the same stuff in remove steps and added a hack for the cases that didnt work that still applies here.

**Comment by stoksc** [comment]
i'll double check though.

**Comment by hkang1** [comment]
> * `webui/react/src/types.ts` ditto, i'll leave webui stuff alone, but cc @hkang1 that sha is going away.

Looked at the specific ones going away. Probably makes sense to maybe hang onto the WebUI type definitions a bit longer in case the older experiment configs persist in the db right? My guess is that the config that gets saved in the experiment table as a JSON blob don't get migrated to eliminate unsupported searcher types.

**Comment by stoksc** [comment]
> My guess is that the config that gets saved in the experiment table as a JSON blob don't get migrated to eliminate unsupported searcher types.

Yeah, no, they don't.

File: docs/tutorials/quick-start.txt
Comment (rb-determined-ai) [inline docs/tutorials/quick-start.txt:310]: somehow I didn't realize until just now how easy to configure adaptive_asha is.  That's so cool.

File: docs/topic-guides/hp-tuning-det/hp-adaptive-asha.txt
Comment (rb-determined-ai) [inline docs/topic-guides/hp-tuning-det/hp-adaptive-asha.txt:171]: Add one word, delete another:

ASHA, or asynchronous halving **algorithm**, ~uses~ asynchronously promotes trials ...

**Review by rb-determined-ai — APPROVED** [review]
Nice work.

File: docs/topic-guides/hp-tuning-det/hp-adaptive-asha.txt
Comment (liamcli) [inline docs/topic-guides/hp-tuning-det/hp-adaptive-asha.txt:59]: SHA -> ASHA

File: docs/topic-guides/hp-tuning-det/hp-adaptive-asha.txt
Comment (liamcli) [inline docs/topic-guides/hp-tuning-det/hp-adaptive-asha.txt:68]: Proposed reorg:
Switch current sections 2 and 3. 

Proposed edit:
The first section here gives a description of the synchronous version of ASHA called successive halving.  The second section discusses the motivation for the asynchronous promotions used by ASHA.  The third section describes why you would choose ``adaptive_asha`` over plain ``asynchronous_halving``.

File: docs/topic-guides/hp-tuning-det/hp-adaptive-asha.txt
Comment (liamcli) [inline docs/topic-guides/hp-tuning-det/hp-adaptive-asha.txt:74]: Successive Halving Algorithm (SHA) and change ASHA->SHA  in this section.

File: docs/topic-guides/hp-tuning-det/hp-adaptive-asha.txt
Comment (liamcli) [inline docs/topic-guides/hp-tuning-det/hp-adaptive-asha.txt:102]: Proposed edit:
In the example above, we generalize "halving" with a field called ``divisor``, which determines what fraction of trials are kept in successive rungs, as well as the training length
in successive rungs.

File: docs/topic-guides/hp-tuning-det/hp-adaptive-asha.txt
Comment (liamcli) [inline docs/topic-guides/hp-tuning-det/hp-adaptive-asha.txt:107]: Let's remove this paragraph.

File: docs/topic-guides/hp-tuning-det/hp-adaptive-asha.txt
Comment (liamcli) [inline docs/topic-guides/hp-tuning-det/hp-adaptive-asha.txt:180]: Please add:
In contrast to SHA which initializes all trials in the bottom rung at the outset, ASHA will continuously add trials to the bottom rung until the desired number of trials is reached.

File: docs/topic-guides/hp-tuning-det/hp-adaptive-asha.txt
Comment (liamcli) [inline docs/topic-guides/hp-tuning-det/hp-adaptive-asha.txt:223]: Proposed edit:
In our experience, early-stopping works well across a variety of deep learning models.  However, there may be some search spaces in which early-stopping underperforms simple random search.  This can happen if model complexity varies drastically in a search space leading to different converge rates or if the search space contains hyperparameters that are strongly correlated with training length.

**Review by liamcli — COMMENTED** [review]
Thanks for doing this!  Let me know if any of my comments are unclear.

File: docs/reference/experiment-config.txt
Comment (stoksc) [reply docs/reference/experiment-config.txt:437]: not sure what this comment is referencing. all the refs should be gone, luckily the docs don't build with dangling refs.

File: docs/reference/experiment-config.txt
Comment (stoksc) [reply docs/reference/experiment-config.txt:436]: yeah, i think that makes sense, i'll do this.

**Comment by stoksc** [comment]
@liamcli I think I hit most of the feedback, save the one I commented about not quite getting.

File: docs/reference/experiment-config.txt
Comment (stoksc) [reply docs/reference/experiment-config.txt]: this is the one i didn't address.

**Comment by stoksc** [comment]
@rb-determined-ai to soothe your conscience, i did check and old experiments still load in the webui.

**Pull request merged**
